artemis-commons/src/main/java/org/apache/activemq/artemis/utils/CompositeAddress.java

```diff
@@ -41,12 +41,36 @@ public static SimpleString toFullyQualified(SimpleString address, SimpleString q
       return result;
    }
 
+   /**
+    * Checks whether a given string complies the fully qualified queue name (FQQN) standard or not.
+    *
+    * @param address The string to check for FQQN standard compliance.
+    * @return boolean
+    */
    public static boolean isFullyQualified(String address) {
-      return address == null ? false : address.contains(SEPARATOR);
+      if (address == null) {
+         return false;
+      }
+
+      // If address length is incapable of holding the separator and two more characters
+      if (address.length() < SEPARATOR.length() + 2) {
+         return false;
+      }
+
+      int index = address.indexOf(SEPARATOR, 1);
+
+      // If separator is not found, at the start or end of string then this is not a FQQN
```

Comment:
Adding " or is " instead of the comma might make the comment clearer. I'd also move the comment above the indexOf call since the result is dependent on that beginning at position 1 in order to carry out the start-of-string check.

Reply:
Also addressed this in the latest commit ;)